### PR TITLE
Updated with TrackCard and TrackDetails component

### PR DESCRIPTION
**Pull request by ihatelactose**
### Description

- Global store to check for the currentlyPlaying track
- Wrote and updated tests for ArtistsContainer
- Updated tests for main component, reducer, and saga for the ArtistsContainer
- Created a separate iTunes API Service and wrote tests for the same
- Changed the main page layout to grid
- updated the route config to point to trackDetails component on track details click
- created the TrackCard and TrackDetails components and wrote tests for the same
- updated the artistsAPI and wrote tests for the same
- created a custom hook useRouter to enable routing, history, location, and other routing based stuff
- updated tests with BrowserRouter that requires access to routing

### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### GIF's
react-challenge-1

### CODE REVIEW

File: app/components/TrackCard/index.js [hunk 1]
```diff
@@ -53,10 +94,13 @@ export function TrackCard({
...
       <If
-        condition={!isEmpty(artistName)}
+        condition={!isEmpty(previewUrl)}
         otherwise={<T data-testid="preview-audio-unavailable" id="preview_audio_unavailable" />}
       >
-        <audio src={previewUrl} controls />
+        <Button data-testid="preview-audio-button" onClick={handleOnPlay} type="primary">
+          {isPlaying ? 'Pause preview' : 'Play Preview'}
+        </Button>
+        <audio data-testid="preview-audio" ref={audioRef} src={previewUrl} />
```
Comment (tousifws) [inline app/components/TrackCard/index.js:103]: Audio controls should show up

File: app/components/TrackCard/index.js [hunk 2]
```diff
@@ -85,6 +129,9 @@ export function TrackCard({
           values={{ releaseDate: new Date(releaseDate).toDateString() }}
         />
       </If>
+      <CustomButton onClick={handleDetailsClick} mt={20}>
+        More Details on this!
```
Comment (tousifws) [inline app/components/TrackCard/index.js:133]: Use react-intl

File: app/components/TrackCard/tests/index.test.js [hunk 3]
```diff
@@ -6,17 +6,27 @@
...
+import { BrowserRouter } from 'react-router-dom';
 
 describe('<TrackCard />', () => {
   it('should render and match with the snapshot', () => {
-    const baseComponent = renderWithIntl(<TrackCard />);
+    const baseComponent = renderWithIntl(
+      <BrowserRouter>
+        <TrackCard />
+      </BrowserRouter>
+    );
```
Comment (tousifws) [inline app/components/TrackCard/tests/index.test.js:20]: Use [MemoryRouter](https://v5.reactrouter.com/web/api/MemoryRouter). Put that in `renderWithIntl`. You won't have to wrap your component with router everywhere.

File: app/components/TrackDetails/index.js [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+      <Container data-testid="track-details" maxwidth={600} padding={20}>
+        <Card data-testid="loading-card">Loading...</Card>
+      </Container>
+    );
+  }
+
+  if (!length) {
+    return (
+      <Container data-testid="track-details" maxwidth={600} padding={20}>
+        <Card data-testid="no-data-card">No such track found.</Card>
```
Comment (tousifws) [inline app/components/TrackDetails/index.js:122]: Use react-intl everywhere. No stray strings

File: app/components/TrackCard/index.js (hunk 1 above)
Comment (ihatelactose) [reply app/components/TrackCard/index.js:103]: The button present above is what controls the controls for the audio element, therefore the controls in the audio element itself are disabled.

File: app/components/TrackDetails/index.js [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+  const router = useRouter();
+  const audioRef = useRef(null);
+  const isMounted = useRef(true);
+
+  useEffect(() => {
+    if (isMounted) {
+      const fetchDetails = async () => {
+        try {
+          const { trackId } = router.query;
+          const data = await getDetails(trackId);
```
Comment (ritwik-wednesday) [inline app/components/TrackDetails/index.js:69]: lets use saga to handle all of our async logic

**Review by tousifws — APPROVED** [review]

**Pull request merged**